src/App.tsx

```diff
@@ -16,80 +17,149 @@ export const goodsFromServer = [
   isReversed: boolean,
+  resetVisible: boolean,
+  classVisible: string,
+  sortBy: string,
+  goods: string[],
 };
 
-// Use this function in the render to prepare goods
-export function getReorderedGoods(
-  goods: string[],
-  { sortType, isReversed }: ReorderOptions,
-) {
-  // To avoid the original array mutation
-  const visibleGoods = [...goods];
+export class App extends React.Component<{}, State> {
+  state = {
+    isReversed: false,
+    resetVisible: false,
+    classVisible: '',
+    sortBy: '',
+    goods: goodsFromServer,
+  };
 
-  // Sort and reverse goods if needed
-  // eslint-disable-next-line no-console
-  console.log(sortType, isReversed);
+  sortAlpha = () => {
+    this.setState({
+      sortBy: 'alpha',
+      resetVisible: true,
+      classVisible: 'ALPABET',
+    });
+  };
 
-  return visibleGoods;
-}
+  sortLength = () => {
+    this.setState({
+      sortBy: 'length',
+      resetVisible: true,
+      classVisible: 'LENGTH',
+    });
+  };
 
-// DON'T save goods to the state
-// type State = {
-//   isReversed: boolean,
-//   sortType: SortType,
-// };
-
-export const App: React.FC = () => {
-  return (
-    <div className="section content">
-      <div className="buttons">
-        <button
-          type="button"
-          className="button is-info is-light"
-        >
-          Sort alphabetically
-        </button>
-
-        <button
-          type="button"
-          className="button is-success is-light"
-        >
-          Sort by length
-        </button>
-
-        <button
-          type="button"
-          className="button is-warning is-light"
-        >
-          Reverse
-        </button>
-
-        <button
-          type="button"
-          className="button is-danger is-light"
-        >
-          Reset
-        </button>
-      </div>
+  reverse = () => {
+    this.setState(state => ({
+      isReversed: !state.isReversed,
+      resetVisible: !state.isReversed,
+    }));
+  };
+
+  reset = () => {
+    this.setState({
+      goods: goodsFromServer,
+      isReversed: false,
+      sortBy: '',
+      resetVisible: false,
+      classVisible: '',
+    });
+  };
+
+  render() {
+    const {
+      isReversed,
+      resetVisible,
+      classVisible,
+      sortBy,
+      goods,
+    } = this.state;
+
+    const visibleGoods = [...goods];
+
+    visibleGoods.sort((alpha, length) => {
+      switch (sortBy) {
+        case 'alpha':
+          return alpha.localeCompare(length);
+
+        case 'length':
+          return alpha[sortBy] - length[sortBy];
+
+        default:
+          return 0;
+      }
+    });
+
+    if (isReversed) {
+      visibleGoods.reverse();
+    }
+
+    return (
+      <div className="section content">
+        <div className="buttons">
+          <button
+            type="button"
+            className={classNames('button', {
+              'is-info is-light': classVisible !== SortType.ALPABET,
+              'is-info': classVisible === SortType.ALPABET,
```

Comment:
```suggestion
              'is-info': sortBy === SortType.ALPABET,
```
це теж можна не тримати в стейті